lib/public_suffix.ex

```diff
@@ -59,13 +59,63 @@ defmodule PublicSuffix do
   @spec registrable_domain(String.t, options) :: nil | String.t
   def registrable_domain(domain, options \\ []) when is_binary(domain) do
     # "The registered or registrable domain is the public suffix plus one additional label."
-    parse_domain(domain, options, 1)
+    parse_domain(domain, options, 1, :registrable_or_suffix)
+  end
+
+  @doc """
+  Parses the provided domain and returns the prevailing rule based on the
+  publicsuffix.org rules. If no rules match, the prevailing rule is "*".
+
+  ## Examples
+
+      iex> prevailing_rule("foo.bar.com")
+      "com"
+      iex> prevailing_rule("co.uk")
+      "co.uk"
+      iex> prevailing_rule("foobar.example")
+      "*"
+
+  You can use the `ignore_private` keyword to exclude private (non-ICANN) domains.
+
+      iex> prevailing_rule("foo.github.io", ignore_private: false)
+      "github.io"
+      iex> prevailing_rule("foo.github.io", ignore_private: true)
+      "io"
+      iex> prevailing_rule("foo.github.io")
+      "github.io"
+  """
+  @spec prevailing_rule(String.t) :: nil | String.t
+  @spec prevailing_rule(String.t, options) :: nil | String.t
```

Comment:
Sure. By the way, I'll also kill the double `@spec` lines if you don't mind. If my understanding is correct there should only be one `@spec` per function, no matter whether an argument is optional or not (still the same arity). E.g. there's no function `public_suffix/1` so `@spec public_suffix(String.t) :: nil | String.t` is wrong. I think...